### PR TITLE
WIP4707: Win64 attribute change

### PR DESCRIPTION
**Pull request by Mike-GC**
New name of attribute "Architecture" with possible values "always32Bit" and "default" was agreed upon between, Rob, Bob, Sean and I during today's WiX Tuesday meeting.

### CODE REVIEW

File: src/libs/WixToolset.Data/Xsd/wix.xsd [hunk 1]
```diff
@@ -7389,6 +7389,24 @@
                     </xs:documentation>
         </xs:annotation>
       </xs:attribute>
+      <xs:attribute name="Architecture">
+        <xs:annotation>
+          <xs:documentation>
+                        Set this attribute to 'always32Bit' to mark this as a 32-bit component. This attribute facilitates
```
Comment (rseanhall) [inline src/libs/WixToolset.Data/Xsd/wix.xsd:7395]: Are there tabs here?

File: src/libs/WixToolset.Data/Xsd/wix.xsd [hunk 2]
```diff
@@ -7389,6 +7389,24 @@
...
+                        the installation of packages that include both 32-bit and 64-bit components. 64-bit
+                        components must not share the same GUID with a 32-bit component.
+          If attribute is default (or specified as 'default'), the architecture set by the -arch switch to candle.exe
```
Comment (rseanhall) [inline src/libs/WixToolset.Data/Xsd/wix.xsd]: I liked your wording better about the default in the other description, this one's a little confusing.

File: src/libs/WixToolset.Data/Xsd/wix.xsd [hunk 3]
```diff
@@ -8755,6 +8760,22 @@
                         </xs:documentation>
             </xs:annotation>
           </xs:attribute>
+          <xs:attribute name="Architecture">
+            <xs:annotation>
+              <xs:documentation>
+                            Set this attribute to 'always32Bit' to mark this as a 32-bit custom action. Valid only when used with
+                                the Script, VBScriptCall, and JScriptCall attributes. The default value is 'default', which
+                                will be the Architecture set by the -arch switch in candle.exe or the InstallerPlatform property
+                                in a .wixproj MSBuild project. The default value is 'default'.
```
Comment (rseanhall) [inline src/libs/WixToolset.Data/Xsd/wix.xsd]: I think you can drop the last sentence, you already explained what the default is.

File: src/tools/wix/Compiler.cs [hunk 4]
```diff
@@ -3451,11 +3451,6 @@ private void ParseCustomActionElement(XElement node)
                 this.core.OnMessage(WixErrors.CustomActionIllegalInnerText(sourceLineNumbers, node.Name.LocalName, innerText, "Script"));
             }
 
-            if (MsiInterop.MsidbCustomActionType64BitScript == (bits & MsiInterop.MsidbCustomActionType64BitScript) && MsiInterop.MsidbCustomActionTypeVBScript != targetBits && MsiInterop.MsidbCustomActionTypeJScript != targetBits)
-            {
-                this.core.OnMessage(WixErrors.IllegalAttributeWithoutOtherAttributes(sourceLineNumbers, node.Name.LocalName, "Win64", "Script", "VBScriptCall", "JScriptCall"));
```
Comment (rseanhall) [inline src/tools/wix/Compiler.cs:3456]: What happened to this error?

File: src/tools/wix/Compiler.cs [hunk 5]
```diff
@@ -3400,7 +3399,8 @@ private void ParseCustomActionElement(XElement node)
                 id = Identifier.Invalid;
             }
 
-            if (!explicitWin64 && (MsiInterop.MsidbCustomActionTypeVBScript == targetBits || MsiInterop.MsidbCustomActionTypeJScript == targetBits) && (Platform.IA64 == this.CurrentPlatform || Platform.X64 == this.CurrentPlatform))
+            // If we're a script CA and we're in a 64-bit package and our Architecture was "default", then set the 64-bit flag
```
Comment (rseanhall) [inline src/tools/wix/Compiler.cs]: The WiX style is for comments to be complete sentences, so they should always be capitalized and have periods.

File: src/libs/WixToolset.Data/Xsd/wix.xsd (hunk 1 above)
Comment (Mike-GC) [inline src/libs/WixToolset.Data/Xsd/wix.xsd:7395]: Just funky spacing. Will be fixed in next push.

File: src/libs/WixToolset.Data/Xsd/wix.xsd (hunk 3 above)
Comment (Mike-GC) [inline src/libs/WixToolset.Data/Xsd/wix.xsd]: A fair point, will be dropped in next push.

File: src/libs/WixToolset.Data/Xsd/wix.xsd (hunk 2 above)
Comment (Mike-GC) [inline src/libs/WixToolset.Data/Xsd/wix.xsd]: Yeah, agreed. Fixing.

File: src/tools/wix/Compiler.cs (hunk 4 above)
Comment (Mike-GC) [inline src/tools/wix/Compiler.cs:3456]: There is no way to explicitly force a 64-bit flag on a 32-bit build, so this build break is now impossible to happen.

**Comment by Mike-GC** [comment]
Thanks for the comments, Sean. Feedback is addressed in latest push.

**Comment by rseanhall** [comment]
We've moved to the .NET Foundation, but we're having trouble getting the bot to check older pull requests.  Here's what it would normally say:

Hi @Mike-GC, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!

In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.dotnetfoundation.org.

TTYL, DNFBOT;

**Comment by robmen** [comment]
This is implemented in WiX v4 now.

**Pull request closed without merging**